src/libstore/build/derivation-goal.cc

```diff
@@ -2097,15 +2094,15 @@ struct RestrictedStore : public LocalFSStore, public virtual RestrictedStoreConf
 
     void registerDrvOutput(const Realisation & info) override
     {
-        if (!goal.isAllowed(info.id.drvPath))
-            throw InvalidPath("cannot register unknown drv output '%s' in recursive Nix", printStorePath(info.id.drvPath));
+        // XXX: Should we check for something here? Probably, but I'm not sure
+        // how
```

Comment:
Probably `registerDrvOutput` shouldn't be allowed in `RestrictedStore` at all, since the caller is completely untrusted and supposed to be sandboxed.

Reply:
I guess once we take the time to properly implement this, we should allow it if the drv output matches an allowed path, but that's complicated to check as we only know its hash modulo and not the drv path. So I'll just forbid everything for now